utils/s3/client.cc

```diff
@@ -223,6 +232,64 @@ future<client::stats> client::get_object_stats(sstring object_name) {
     co_return st;
 }
 
+future<client::TagSet> client::get_object_tagging(sstring object_name) {
+    // see https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectTagging.html
+    auto req = http::request::make("GET", _host, fmt::format("{}?tagging", object_name));
+    s3l.trace("GET {} tagging", object_name);
+    TagSet tagging;
+    co_await make_request(std::move(req),
+                          [&tagging] (const http::reply& reply, input_stream<char>&& in) mutable -> future<> {
+        auto& tagset = tagging;
+        auto input = std::move(in);
+        auto body = co_await util::read_entire_stream_contiguous(input);
+        auto doc = std::make_unique<rapidxml::xml_document<>>();
```

Comment:
Please, put parsing into a dedicated helper function next to `parse_multipart_upload_id` and `parse_multipart_copy_upload_etag` so that we could collect all XML-bodies parsing in one place some day

Reply:
> Please, put parsing into a dedicated helper function

i put the parsing into a helper.

> next to parse_multipart_upload_id and parse_multipart_copy_upload_etag so that we could collect all XML-bodies parsing in one place some day

why do we want to put them together?  i am not sure it's better to group the helpers together is an improvement though. for two reasons:

1. `parse_multipart_upload_id()` and `parse_multipart_copy_upload_etag()` are helpers for the methods of `client::upload_sink_base`. this class is in turn used by `client::make_upload_sink()` and `client::make_upload_jumbo_sink()`. i think that's why they are grouped together. for the same reason, `client::readable_file` and `client::make_readable_file()` are co-located. this helps with the readability. as the declarations and definitions designed as a part of the similar functionality are grouped together.  no need to jump back and forth when reading the source code. he or she can just read through them. the same applies to `parse_http_last_modified_time()` and `client::get_object_stats()`. if we group the parsers and printers together, it would hurt the readability and maintainability.
2. from a more practical perspective, we don't need to have a forward declarations of the helper or place the definitions of helpers before their consumer. please note, i intentionally put the definition of the tagging related function in the same order of their declarations. and the methods' declarations are ordered from the minimal stats to an object's attributes, and to its data.